File: work-backups/scatmpi.c
/*      program scatmpi.c */
/* Illustrates mpi_scatter. */
#include <stdio.h>
/* Includes the mpi C library. */
#include "mpi.h"
      main(int argc, char* argv[]) {
      float a_list[7],a_loc[7];
      int my_rank,p,n,dest,tag,i;
      n    = 1024;
      dest = 0;
      tag  = 50;
/* Initializes mpi, gets the rank of the processor, my_rank, */
/* and number of processors, p. */
      MPI_Init(&argc, &argv);
      MPI_Comm_rank(MPI_COMM_WORLD,&my_rank);
      MPI_Comm_size(MPI_COMM_WORLD,&p);
      if (my_rank==0) {
        for (i = 0;i<8;i++) {
          a_list[i] = i; 
        }
      }
/* The array, a_list, is sent and recieved in groups of */
/* two to the other processors and stored in the arrays a_loc. */
      MPI_Scatter(&a_list,2,MPI_FLOAT,&a_loc,2,MPI_FLOAT,0,
                        MPI_COMM_WORLD);
      for (i = 0; i<2; i++) {
        printf("my_rank = %d i = %d a_loc = %f\n",my_rank,i,a_loc[i]);
      }
/* mpi is terminated. */ 
      MPI_Finalize();
}  /* end program scatmpi */

File: work-backups/scatmpi.f
      program scatmpi
      implicit none
      include 'mpif.h'
      real, dimension(0:7):: a_list,a_loc
      integer::  my_rank,p,n,source,dest,tag,ierr,loc_n
      integer:: i,status(mpi_status_size)
      data n,dest,tag/1024,0,50/
      call mpi_init(ierr)
      call mpi_comm_rank(mpi_comm_world,my_rank,ierr)
      call mpi_comm_size(mpi_comm_world,p,ierr)
      if (my_rank.eq.0) then
        do i = 0,7
        	a_list(i) = i
        end do
      end if
      call mpi_scatter(a_list,2,mpi_real,a_loc,2,mpi_real,0,&
                        mpi_comm_world,status,ierr)
      print*, 'my_rank =',my_rank,'a_loc = ', a_loc
      call mpi_finalize(ierr)
      end program scatmpi

